File: jscomp/test/test_http_server.ml
let port = 3000
let hostname = "127.0.0.1"

let create_server http =
  let server =
    http##createServer (fun [@bs] req resp ->
        resp ## statusCode #= 200 ;
        resp##setHeader "Content-Type" "text/plain" ;
        resp##_end "Hello world\n") in
  server##listen port hostname (fun [@bs] () ->
      Js.log
        ( "Server running at http://" ^ hostname ^ ":" ^ string_of_int port
        ^ "/" ))

let () = create_server Http_types.http

File: jscomp/test/test_http_server.js
'use strict';

var Http = require("http");

var hostname = "127.0.0.1";

function create_server(http) {
  var server = http.createServer((function (_, resp) {
          resp.statusCode = 200;
          resp.setHeader("Content-Type", "text/plain");
          return resp.end("Hello world\n");
        }));
  return server.listen(3000, hostname, (function () {
                console.log("Server running at http://" + (hostname + (":" + (String(3000) + "/"))));
                return /* () */0;
              }));
}

create_server(Http);

/*  Not a pure module */
